### ISSUE TITLE
o-forms: date inputs visual and accessible label names do not match - ID: DAC_Label_In_Name_01

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#527]
# Issue
This is from page 80 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=inverse-form&system_code=origami-repo-data&brand=master#my-date-input>

The accessible name of the date inputs does not contain the string that is presented visually as the label ‘DD / MM / YYYY’. Voice activation users may be unable to reference the component to their assistive technology by speaking the visual label, as this does not correspond with the accessible name of the component. Aria-hidden has also been used on the visible labels which removes the labels from the accessibility tree.
 
# Proposed Solution:
For user interface components with labels that include text or images of text, the accessible name should contain the text that is presented visually within the label. Please note that best practice is to have the text of the label at the start of the name. It may be advisable to use native HTML including visually hidden text over WAI-ARIA where possible. For more information, please refer to [F96: Failure due to the accessible name not containing the visible label text.](https://www.w3.org/WAI/WCAG21/Techniques/failures/F96)

Remove the aria-label and aria-hidden attributes and use an explicit label with matching valid ‘for’ and ‘id’ values. This will ensure that the specified associated `<label>` element acts as the accessible name.

```html
<input type=”text” id=”date” name=”invalid-date” value=”01” class=”o-forms-input__day-part” inputmode=”numeric” pattern=”0[1-9]|[12]\d|3[01]” required=””>
<span class=”o-forms-input__label> <label for=”date”>Day</label>
```

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=date&system_code=origami-repo-data&brand=master> and found on page 101 of the audit report

**Comment by notlee** [comment]
We'll need some design input to solve this problem

**Comment by notlee** [comment]
Ako already did this!
https://github.com/Financial-Times/origami/commit/c3ccb8184b65f98315f3567ee1aa3e4b95c9d3fd
